reconcile/ocm_oidc_idp.py

```diff
@@ -0,0 +1,193 @@
+import logging
+import sys
+from collections.abc import Callable
+from typing import (
+    Any,
+    Iterable,
+    Sequence,
+)
+
+from reconcile import queries
+from reconcile.gql_definitions.ocm_oidc_idp.clusters import (
+    ClusterAuthOIDCV1,
+    ClusterV1,
+)
+from reconcile.gql_definitions.ocm_oidc_idp.clusters import query as cluster_query
+from reconcile.ocm.types import OCMOidcIdp
+from reconcile.ocm.utils import cluster_disabled_integrations
+from reconcile.status import ExitCodes
+from reconcile.utils import gql
+from reconcile.utils.ocm import OCMMap
+from reconcile.utils.secret_reader import SecretReader
+
+QONTRACT_INTEGRATION = "ocm-oidc-idp"
+
+DEFAULT_EMAIL_CLAIMS: list[str] = ["email"]
+DEFAULT_NAME_CLAIMS: list[str] = ["name"]
+DEFAULT_USERNAME_CLAIMS: list[str] = ["username"]
+DEFAULT_GROUPS_CLAIMS: list[str] = []
+
+
+def fetch_current_state(
+    ocm_map: OCMMap, clusters: Iterable[ClusterV1]
+) -> list[OCMOidcIdp]:
+    """Fetch all current configured OIDC identity providers."""
+    current_state = []
+
+    for cluster in clusters:
+        ocm = ocm_map.get(cluster.name)
+        current_state += ocm.get_oidc_idps(cluster.name)
+
+    return current_state
+
+
+def fetch_desired_state(
+    secret_reader: SecretReader,
+    clusters: Iterable[ClusterV1],
+    vault_input_path: str,
+) -> list[OCMOidcIdp]:
+    """Compile a list of desired OIDC identity providers from app-interface."""
+    desired_state = []
+    error = False
+    for cluster in clusters:
+        for auth in cluster.auth:
+            if not isinstance(auth, ClusterAuthOIDCV1):
+                continue
+
+            secret = {
+                "path": f"{vault_input_path.rstrip('/')}/{QONTRACT_INTEGRATION}/{auth.name}/{cluster.name}"
+            }
```

Comment:
again for future-refactoring-you - opportunity to have a common function to build the secret path across all idp provider integrations

Reply:
Let's keep this for later when `ocm-github-idp` will be refactored.